llrt_core/src/modules/http/fetch.rs

```diff
@@ -131,6 +134,50 @@ pub(crate) fn init(ctx: &Ctx<'_>, globals: &Object) -> Result<()> {
     Ok(())
 }
 
+fn parse_data_url<'js>(ctx: &Ctx<'js>, data_url: &String) -> Result<Response<'js>> {
+    let (mime_type, data) = data_url
+        .trim_start_matches("data:")
+        .split_once(',')
+        .ok_or_else(|| Exception::throw_type(ctx, "Invalid data URL format"))?;
+
+    let (is_base64, mime_type) = if mime_type.contains("base64") {
```

Comment:
Could we simplify this? split_once by ";" and check if second part start with `base64`?

Reply:
`base64` does not always come in second. If `charset` is attached to `mime_type`, it may come third because `;` is used as a separator. If we start considering the second or third, the code is going to be redundant...

This method is basically the same as described in the [Fetch Standard](https://fetch.spec.whatwg.org/#data-urls), with a few modifications.

Reply:
I think a better approach here would be to do something like this:
```rust
  let mut is_base_64 = false;
  let mut content_type = String::with_capacity(10);
  for (i, part) in data.split(";").enumerate(){
      let part = part.trim();
      if  i == 1 || i == 2 {
          if part == "base64" {
              is_base_64 = true;
              break;
          }
          content_type.push(';');
      }
      content_type.push_str(part)
  }
```

Reply:
Thank you. It seems to be working well in several patterns that I can think of. :)

Reply:
Thank you. It seems to be working well in several patterns that I can think of. :)